<h1 id="self-guided-tour-of-databases">Self guided tour of databases</h1>

<p>MG-RAST/ SRA / SiLVA / greengenes / RDP / VAMPS / PHAST / UNITE</p>

<p>Author: Tracy Teal</p>

<h2 id="objectives">Objectives</h2>

<ul>
<li>Gain familiarity with some common databases</li>
<li>What questions might you use this database to address</li>
<li>How would you get data in</li>
<li>Can you interact with data on the site</li>
<li>How can you get data out</li>
<li>Is the &#39;help&#39; helpful</li>
<li>How can you get data in/out/process for more than one sample/sequence</li>
</ul>

<h2 id="the-tour">The Tour</h2>

<h3 id="introduction">Introduction</h3>

<p>Welcome to the self-guided tour of genomic/metagenomic databases. We know
that you&#39;re in for a fun ride ahead! Think of this as a walking tour / choose
your own adventure. Take your time and try different paths if you figure
out you chose one that led you to perish in a dark cave or you just
wanted to try a different route.</p>

<p>The below set of questions are aimed to help you explore the database
and figure out how to interact with it. Use as you wish.</p>

<h3 id="go-to-the-website">Go to the website</h3>

<p>Probably the database has some obvious URL. You should be able to find it
just by searching for it though.</p>

<h3 id="accounts">Accounts</h3>

<ul>
<li><p>Do you have to create an account to interact with this site? Would creating
an account give you access to features you wouldn&#39;t otherwise have? Why might
a database ask you to create an account?</p></li>
<li><p>If there is an ability to create an account and it doesn&#39;t request your first
born as a part of that process, go ahead and create an account. Was there
an approval process? Did it ask a lot of questions? Was it too terrible?</p></li>
</ul>

<p>Often the need to create an account is the biggest barrier to new people using
a site. They don&#39;t want to create an account before they know it will be useful
to them, or the process is time consuming and is not deemed to be worth it.</p>

<h3 id="data-availability">Data availability</h3>

<ul>
<li>Is there publicly available or example data that you can use to explore the
features of the site?</li>
</ul>

<h3 id="uploading-data">Uploading data</h3>

<ul>
<li>If this database takes new data, what format of data does it require? Do
you know how to provide data in this format? If they have particular guidelines
is the &#39;help&#39; on data format clear and helpful? Are there templates?</li>
</ul>

<p>Another of the biggest barriers to working with tools is data format. If a
site asks for data in a format that is not generally created (FASTA, FASTQ,BIOM)
then you will have to figure out how to get it in that format. Sometimes that
requires finding a converter. The site might have one or have good documentation
on how to do it. Or you might need to write your own script. This is
called &quot;data wrangling&quot;. It&#39;s not particularly fun, but it&#39;s a part of
many genomic/metagenomic processes.</p>

<h3 id="questions">Questions</h3>

<p>Formulate a question that you might ask using the data or features of
this site.</p>

<ul>
<li><p>If there is the ability to analyze or explore data, try to
answer that question on the web site.</p></li>
<li><p>If the site allows for or requires downloading data, how would you download
the data to address your question?</p></li>
</ul>

<h3 id="downloading-data">Downloading data</h3>

<ul>
<li>How would you download data from this site? What data format does
the data come in once you download it? Is this a humungous download?
Is it reasonable to download the data this way?</li>
</ul>

<h3 id="alternative-ways-of-interacting-with-the-site">Alternative ways of interacting with the site</h3>

<ul>
<li><p>Is there another way to upload/download/analyze data for this database
other than through the web interface? If so, is there good documentation
on how to do so at the site? If there&#39;s not, is that documentation
somewhere else? Could you Google for it?</p></li>
<li><p>If you can access data another way, can you figure out how to do it?</p></li>
</ul>

<h3 id="help">Help</h3>

<ul>
<li>If you haven&#39;t already, go to the Help, FAQ or Forum section of the web
site. Is there helpful &#39;help&#39; documentation? Is there an active user
community that could help address issues or questions you might have?</li>
</ul>

<h3 id="automation">Automation</h3>

<ul>
<li>In the above sections, you worked to address some questions, probably
for one sample or sequence? How would it be to try to do this for more
than one sample/sequence or even hundreds or thousands? Is there a more
automated way to do this, or does it all need to be done &#39;by hand&#39;?</li>
</ul>

<h3 id="overall-experience">Overall experience</h3>

<p>You made it to the end! You have your data / answered your question / uploaded
your data for archival purposes. Congratultions!</p>

<ul>
<li><p>How was the experience? Would you want to use this database again?</p></li>
<li><p>If this was how your data was provided to other researchers would you
be happy? Could others easily access your data?</p></li>
<li><p>Are there other common alternatives to this database?</p></li>
<li><p>Do you have thoughts on what could change? If it&#39;s not a huge database,
developers often appreciate constructive feedback. It could be worth
filing an issue or making a comment, especially if something isn&#39;t working.</p></li>
<li><p>Bonus question: Are there other alternatives to this database? Maybe ones
that aren&#39;t as specific to this type of data? Dryad, Figshare</p></li>
<li><p>Bonus question: If this is a database that takes data and shares it
publicly, how are you getting credit for that data? Is there a DOI? Something
else?</p></li>
<li><p>Bonus question: What happens to the data if this database goes away? How
do you think this database is funded?</p></li>
</ul>
