<hr>

<p>layout: page
title: &quot;MacOS Users: Installing MacQIIME&quot;
category: tut
date: 2014-08-12 08:55:36</p>

<h2 id="order:-2">order: 2</h2>

<h2 id="macosx-users:-installing-macqiime-(and-additional-packages)">MacOSX users: Installing MacQIIME (and additional packages)</h2>

<p>These directions are slightly modified from <a href="http://www.wernerlab.org/software/macqiime/macqiime-installation">those provided by the Werner lab</a>.</p>

<p>Please note:
* You must have administrator access on a computer to install MacQIIME.
* You must install R and the required dependency packages.  We recommend that you install R before installing MacQIIME.</p>

<h3 id="installing-macqiime">Installing MacQIIME</h3>

<ol>
<li>
Download the tgz archive: [http://www.wernerlab.org/software/macqiime](http://www.wernerlab.org/software/macqiime).  The current version of QIIME is 1.8.0.  Save the archive (do not open with Archive Utility).  It should automatically be saved in the Downloads directory.

![IMG1](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG1.png)
</li>

<li>
Open your computer's terminal. <strong>Applications -> Utilities -> Terminal</strong>
</li>
<li>
Navigate to the Downloads directory (where the archive was saved), using the `cd` (change directory) command:

`cd ~/Downloads`
</li>
<li>
Check that the file was not corrupted using the “md5” command (message digest 5).  This command provides the sum of the “hashes” to determine the size of the file.  The idea is that if the file were corrupted during download (missing some pieces), the output will be different.  See the “MD5 hashes” section of [this Wikipedia article for an example](http://en.wikipedia.org/wiki/MD5):

To do this, you would type:

`md5 MacQIIME_1.8.0-20140103_OS10.6.tgz`

The computer should return the following:

`MD5 (MacQIIME_1.8.0-20140103_OS10.6.tgz) = 9f1175e0a0ba71ebe6bb5d9b7bb0e6c2`

Check carefully that the number above matches exactly your output.  If it does not, discard the downloaded file (move to trash, empty trash), and start over with an uncorrupted or complete download.
</li>
<li>
Now you will want to unarchive using the tar command:

`tar -xvf MacQIIME_1.8.0-20140103_OS10.6.tgz`

This will show a list of file contents as they are extracted from the archive:

![IMG2](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG2.png)

Unarchiving the file will automatically create a new directory called “MacQIIME_1.8.0-20140103_OS10.6”.  Use `cd` to move into that directory:

`cd MacQIIME_1.8.0-20140103_OS10.6/`
</li>
<li>
Use the “ls” command (list) to view the files in that directory:

`ls`

The computer will return the following:

`README.txt install.s   macqiime    scripts`

Run the install.s script to install most of what is needed to run QIIME. Use the command:

`./install.s`

Then, the script will prompt you to agree to write-over the old directory.

![IMG3](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG3.png)

As we are starting from scratch this is okay, and you may hit enter.  In the future, if you upgrade to a new QIIME version, it is recommend that you save the older version before upgrading.  [There are directions for saving the old version of MacQIIME](http://www.wernerlab.org/software/macqiime/macqiime-installation).

Then, you will be prompted for a password.  This is where administrator access (`sudo`) is needed:

![IMG4](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG4.png)

After you type in your password, and the script will do its thing!

At the end, the terminal look like:

![IMG5](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG5.png)
</li>
</ol>

<h3 id="installing-the-fastq-join-and-seqprep-tools">Installing the fastq-join and SeqPrep tools</h3>

<p>This MacQIIME add-on is for joining paired-end Illumina reads.
<ol>
<li>
Download the tgz archive:
<a href="http://www.wernerlab.org/software/macqiime/add-fastq-join-to-macqiime-1-8-0">http://www.wernerlab.org/software/macqiime/add-fastq-join-to-macqiime-1-8-0</a>.
</li>
<li>
Save the archive (do not open with Archive Utility).  The archive (“Add_ea-tools_to_MacQIIME-1.8.0.tgz”) should automatically be saved in the Downloads directory.</p>

<p><img src="https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG6.png" alt="IMG6">
</li>
<li>
In terminal, navigate back to the Downloads directory by typing:</p>

<p><code>cd ~/Downloads/</code>
<li>
As before, unarchive:</p>

<p><code>tar -xvf Add_ea-tools_to_MacQIIME-1.8.0.tgz</code></p>

<p><img src="https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG7.png" alt="IMG7">
<li>
Then, use “cd” to navigate into the new directory:</p>

<p><code>cd Add_ea-tools_to_MacQIIME-1.8.0/</code>
</li>
<li>
Run the installation script:</p>

<p><code>./install.script</code></p>

<p>You will again be prompted for the administrative password.  There is no special message that is returned when the installation is complete, but you will know by the “$” at the end of your userid, which means that the computer is ready for your next command.</p>

<p><img src="https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG8.png" alt="IMG8">
</li>
</ol></p>

<h3 id="installing-legacy-blast-2.2.22">Installing legacy BLAST 2.2.22</h3>

<ol>
<li>
Download (but do not open) the legacy BLAST 2.2.22 archive by copying and pasting the following link into your web browser:

`ftp://ftp.ncbi.nlm.nih.gov/blast/executables/release/2.2.22/blast-2.2.22-universal-macosx.tar.gz`

![IMG9](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG9.png)

The file should be automatically saved into your Downloads directory.
</li>

<li>
Return to your terminal.  As before, navigate to the Downloads directory:

`cd ~/Downloads`

Then, unarchive using the “tar” command (do you sense a pattern here?):

`tar -xvf blast-2.2.22-universal-macosx.tar.gz`

A long list of files will be displayed on the screen as they are “unpacked” from the archive.  A new directory called “blast-2.2.22” will be automatically created.
</li>

<li>
Copy (“`cp`”) the new blast directory to the opt (options) directory of the computer.   You may be prompted to provide your password.

`sudo cp -R blast-2.2.22 /opt/`
</li>

<li>
Now, we have to add blast-2.2.22 to the PATH so that the computer knows were to find the BLAST scripts and files.  To do this, we must open an invisible file called “bashrc” using an editor called “nano.”

`sudo nano /etc/bashrc`

The nano editor will open the file, and it will look something like this:

![IMG10](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG10.png)

Then, you want to copy and past the following at the end (after “fi”):

`export PATH=/opt/blast-2.2.22/bin:${PATH}`

![IMG11](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG11.png)

Save the file by using Control-O (“write-out”), and then hit return to accept the file name (essentially, writing over the old bashrc).  Then exit the nano editor using Control-X.
</li>

<li>
Close the terminal window and start a new one.  This is to allows the computer re-reading the bashrc file that we edited.  To determine whether the computer can find an executable file(blastall) in the directory blast-2.2.22 by reading the bashrc file, use the “which” command:

`which blastall`

The computer will return the location of the of the blastall file if it is now correcly in your path.

![IMG12](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG12.png)
</li>

<li>
Now, we must make another file that points the computer to the BLAST data directory.  This file will be called ncbirc.  We will use the nano editor again for this.

`nano ~/.ncbirc`

Your computer will return an empty file, because we are creating it right now!

![IMG13](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG13.png)

Copy and paste this text into the empty file:

`Data=/opt/blast-2.2.22/data/`

![IMG14](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG14.png)

Save our shiny new `.ncbirc` file using Control-O, press return to accept the file name, and then exit the nano editor using Control-X.

*Note:* If there is more than one user on a computer, the `.ncbirc` file will have to be created independently for every user.   To do this, repeat step 6 within every user profile.
</li>
</ol>

<h3 id="testing-macqiime">Testing MacQIIME</h3>

<ol>
<li>
To check if the rest of QIIME was installed correctly, open a new terminal window and type “macqiime” and hit return.  Now you are in the “QIIME” environment, with access to all of its analysis goodies.

You must type macqiime at the beginning of every new terminal session during which you want to use QIIME.

`macqqime`

Then, use the command:

`print_qiime_config.py –t`

This will help you to determine which versions of all of the dependencies are running on MacQIIME, and it will also test to make sure they are working correctly.  The computer will print the results of the tests.

![IMG15](https://raw.githubusercontent.com/edamame-course/docs/gh-pages/img/macqiime/IMG15.png)

You will note some tests “fail”, such as AmpliconNoise and usearch; this makes sense because we chose not to stall these add-ons.
</li>
</ol>

<h3 id="extra-add-ons-to-qiime">Extra Add-Ons to QIIME</h3>

<p>For the purposes of the workshop, we will not be installing or using:
* <a href="https://code.google.com/p/ampliconnoise/">AmpliconNoise</a> (alternative to denoise.py, which is installed with MacQIIME)
* <a href="http://www.drive5.com/usearch/">usearch</a> (alternative OTU picking and clustering methods available)
* <a href="http://sourceforge.net/projects/sourcetracker/">SourceTracker</a> (Bayesian classification of samples to their likely source)
* <a href="http://topiaryexplorer.sourceforge.net/">Topiary Explorer</a> (for visualization of sample metadata onto phylogenetic trees)
* <a href="http://www.cytoscape.org/">Cytoscape</a> (for creating OTU networks)</p>

<p>Feel free to explore these other packages for your own data analysis.</p>
