<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="viewport" content="width=device-width">

        <title>EDAMAME : Metagenome Annotation</title>
        <meta name="description" content="course materials & tutorials">

        <link href="//netdna.bootstrapcdn.com/twitter-bootstrap/2.3.2/css/bootstrap-combined.no-icons.min.css" rel="stylesheet">
        <link rel="stylesheet" href="/css/syntax.css">
        <link rel="stylesheet" href="/css/main.css">
    </head>
    <body>

        <div class="container">
            <div class=row-fluid>
                <div id=header class=span12>
                    <h4><a class=brand href="http://edamame-course.org">EDAMAME</a>
    <small>course materials & tutorials</small>
</h4>

                </div>
            </div>

            <div class=row-fluid>
                
                
                    <div id=navigation class=span3>
                        
<ul class="nav nav-list">
    <li class=nav-header>HOME</li>
    <li><a href="index.html">EDAMAME course materials</a></li>
    <li class=divider></li>
    <li class=nav-header>Before We Begin</li>
    <li><a href="https://edamame-course.github.io/docs/general_set_up_guidelines.html">Getting Set Up For The Course</a></li>
    <li><a href="https://edamame-course.github.io/docs/intro_to_ec2_instance.html">All Users: Setting up an EC2 instance</a></li>
    <li><a href="https://edamame-course.github.io/docs/mobaxterm.html">Windows Users: Connecting to your AWS EC2 instance</a></li>
    <li><a href="https://edamame-course.github.io/docs/mac_os_linux_connect_to_ec2.html">Mac and Linux Users: Connecting to your AWS EC2 instance</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/macqiime_installation.html">Mac Users: Installing MacQIIME</a></li>
    <li><a href="https://edamame-course.github.io/docs/QIIME_VB_for_Windows.html">Windows Users: Installing a Virtual Box to run QIIME on your own Windows computer</a></li>
    <li class=divider></li>
    <li class=nav-header>Wednesday, August 13th</li>
    <li><a href="https://edamame-course.github.io/docs/august_13.html">Wednesday, August 13th Overview</a></li>
    <li><a href="https://edamame-course.github.io/docs/the_shell.html">An Introduction To Using The Shell</a></li>
    <li><a href="https://edamame-course.github.io/docs/intro_qiime_tutorial.html">Introduction to Data Analysis with QIIME</a></li>
    <li class=divider></li>
    <li class=nav-header>Thursday, August 14th</li>
    <li><a href="https://edamame-course.github.io/docs/august_14.html">Thursday, August 14th Overview</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-08-14-AM_Ashley_Lecture2.pdf?raw=true">Alpha Diversity Lecture</a></li>
    <li><a href="https://edamame-course.github.io/docs/qiime_2_tutorial.html">QIIME Tutorial Part Two</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-08-14-PM_Ashley_Lecture3.pdf?raw=true">Beta Diversity Lecture</a></li>
    <li><a href="https://edamame-course.github.io/docs/qiime_3.html">QIIME Tutorial Part Three</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-edamame-assembly_TBrown.pdf?raw=true">Titus Brown's Intro to Metagenomics Lecture</a></li>
    <li class=divider></li>
    <li class=nav-header>Friday, August 15th</li>
    <li><a href="https://edamame-course.github.io/docs/august_15.html">Friday, August 15th Overview</a></li>
    <li><a href="https://edamame-course.github.io/docs/intro_to_ec2_instance.html">Introduction to using Amazon EC2</a></li>
    <li><a href="http://pschloss.github.io/2014_08_15-mothur/#1">Pat Schloss's Presentation</a></li>
    <li><a href="http://www.mothur.org/wiki/MiSeq_SOP">mothur to the max</a></li>
    <li class=divider></li>
    <li class=nav-header>Saturday, August 16th</li>
    <li><a href="https://edamame-course.github.io/docs/august_15.html">Saturday, August 16th Overview</a></li>
    <li><a href="">Introduction to Metagenomics</a></li>
    <li><a href="">Using BLAST at the command line</a></li>
    <li class=divider></li>
    <li class=nav-header>Sunday, August 17th</li>
    <li>Brunch with Jim Tiedje!</li>
    <li><a href="">Metagenome Assembly</a></li>
    <li class=divider></li>
    <li class=nav-header>Monday, August 18th</li>
    <li><a href="https://github.com/edamame-course/docs/august_18.html">Monday, August 18th Overview</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-18-AM_Ashley_Lecture4.pdf?raw=true">Hypothesis testing, identifying gradients, and linking environmental and community data</a></li>
    <li><a href="https://github.com/edamame-course/docs/tree/gh-pages/extra/Jones_R_Tutorial_Files">Using R for community ecology analyses</a></li>
    <li><a href=""></a></li>
    <li class=divider></li>
    <li class=nav-header>Tuesday, August 19th</li>
    <li><a href="">Jim Cole - Guest Lecture</a></li>
    <li><a href="http://adina-howe.readthedocs.org/en/latest/">Adina Howe: Tutorial on working with NCBI and MG-RAST and their APIs</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-19-Docherty_NEON.pdf?raw=true">Kathryn Docherty - Guest Lecture</a></li>
    <li><a href="https://edamame-course.github.io/docs/metagenome_assembly_and_annotation.html">Josh: Metagenome Assembly, Annotation, and Quality Assessment</a></li>
    <li><a href="https://edamame-course.github.io/docs/using_dropbox_on_your_ec2_machine.html">Josh: Installing Dropbox on any computer from the command line</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/_posts/2014-08-19_database_questions.md">Self Guided Tour of Databases</a></li>
    <li><a href="">Josh speaks about the wild and woolly world of other marker regions!</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-19-Peralta.pdf?raw=true"></a>Guest lecture from Ariane Peralta on Human-impacted ecosystems</li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-20-metagenomic_visualization.pdf?raw=true">Tracy gives a tour of data visualization!</a></li>
    <li class=divider></li>
    <li class=nav-header>Wednesday, August 20th</li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-08-20_Cluster_Computing_EDAMAME.pdf?raw=true">Guest Lecture from Dirk Colbry - Using a Computer Cluster</a></li>
    <li>Course Wrap-Up and Evaluation</li>
</ul>

                    </div>

                    <div id=content class=span9>
                        <div class=page-header>
    <h2>Metagenome Annotation
        
    </h2>
</div>

<h4 id="1.-assembling"><strong>1.</strong> Assembling</h4>

<p>At last!  All that filtering and diginorming is done, and we can get
down to the serious business of assembling.  Huzzah!</p>

<p>Let&#39;s start with some collection of group files in <code>/mnt/assembly</code>.</p>

<h5 id="install-the-spades-assembler">Install the spades Assembler</h5>

<p>Let&#39;s try assembling the sequences with the <a href="http://bioinf.spbau.ru/spades/">SPADes</a>.</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">apt-get -y install cmake
</code></pre></div>
<p>then:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">wget http://spades.bioinf.spbau.ru/release2.5.1/SPAdes-2.5.1.tar.gz
tar -xzf SPAdes-2.5.1.tar.gz
cd SPAdes-2.5.1
PREFIX=/usr/local ./spades_compile.sh
</code></pre></div>
<h5 id="setting-up-the-spades-run">Setting Up The SPAdes run</h5>

<p>One of the reasons we are using SPAdes is because it runs an assembly on a set of k-mer values automatically.  It also has a simple structure.</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">spades.py -s sequences.fastq -o assembly
</code></pre></div>
<p>This will run for a while!  (We&#39;ve already done this and will look at the assembly now).</p>

<h4 id="getting-stats-for-the-assemblies">Getting Stats For the Assemblies</h4>

<p>To get some basic stats for the assemblies, run:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">python /usr/local/share/khmer/sandbox/assemstats3.py 1000 assembly/scaffolds.fasta
</code></pre></div>
<p>This will yield something like:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">N       sum     max     filename
38      671957  83467   dn.21/contigs.fa
32      668918  83568   dn.23/contigs.fa
35      668509  83401   dn.25/contigs.fa
31      671843  83817   dn.27/contigs.fa
32      669104  83721   dn.29/contigs.fa
32      672735  84066   dn.31/contigs.fa
32      673102  83774   dn.33/contigs.fa
31      674629  83912   dn.35/contigs.fa
31      677446  84200   dn.37/contigs.fa
33      681099  84554   dn.39/contigs.fa
35      685245  84852   dn.41/contigs.fa
40      686733  85276   dn.43/contigs.fa
41      649574  62719   nodn.21/contigs.fa
39      639388  62155   nodn.23/contigs.fa
49      646132  62145   nodn.25/contigs.fa
39      647100  83798   nodn.27/contigs.fa
38      650487  83750   nodn.29/contigs.fa
33      649863  83770   nodn.31/contigs.fa
31      636979  83822   nodn.33/contigs.fa
35      645536  83856   nodn.35/contigs.fa
36      647848  83800   nodn.37/contigs.fa
33      654660  83934   nodn.39/contigs.fa
36      645126  83897   nodn.41/contigs.fa
34      660289  83231   idba.dn.d/scaffold.fa
45      666147  41120   idba.nodn.d/scaffold.fa
</code></pre></div>
<h4 id="2.-annotating-your-metagenome-with-prokka"><strong>2.</strong> Annotating your metagenome with Prokka</h4>

<h5 id="installing-prokka">Installing Prokka</h5>

<p>I like the <a href="http://www.vicbioinformatics.com/software.prokka.shtml">Prokka Annotation Software Program</a> to annotate your assembled metagenome.</p>

<p>We have to download and install a lot of stuff, though -- estimated ~15-20 minutes.</p>

<p>First, we need to install BioPerl and NCBI BLAST+; because we&#39;re using an Amazon EC2 ubuntu machine, we&#39;ll use the Debian Linux package installer <code>apt-get</code>.</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">apt-get update
apt-get -y install bioperl ncbi-blast+
</code></pre></div>
<p>Now you want to download and unpack Prokka:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O http://www.vicbioinformatics.com/prokka-1.7.tar.gz
tar xzf prokka-1.7.tar.gz 
curl -O http://www.vicbioinformatics.com/prokka-1.7.2
cp prokka-1.7.2 prokka-1.7/bin/prokka
</code></pre></div>
<p>Prokka depends on a lot of other software, too; so we&#39;ll need to install those tools:</p>

<p>Install <a href="http://hmmer.janelia.org">HMMER</a>:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O ftp://selab.janelia.org/pub/software/hmmer3/3.1b1/hmmer-3.1b1.tar.gz
tar xzf hmmer-3.1b1.tar.gz 
cd hmmer-3.1b1/
./configure --prefix=/usr &amp;&amp; make &amp;&amp; make install
</code></pre></div>
<p>Install <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC373265">Aragorn</a>:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O http://mbio-serv2.mbioekol.lu.se/ARAGORN/Downloads/aragorn1.2.36.tgz
tar -xvzf aragorn1.2.36.tgz
cd aragorn1.2.36/
gcc -O3 -ffast-math -finline-functions -o aragorn aragorn1.2.36.c
cp aragorn /usr/local/bin
</code></pre></div>
<p>Install <a href="http://prodigal.ornl.gov">Prodigal</a>:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O http://prodigal.googlecode.com/files/prodigal.v2_60.tar.gz
tar xzf prodigal.v2_60.tar.gz 
cd prodigal.v2_60/
make
cp prodigal /usr/local/bin
</code></pre></div>
<p>Install <a href="http://www.ncbi.nlm.nih.gov/genbank/tbl2asn2">tbl2asn</a>:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools/converters/by_program/tbl2asn/linux64.tbl2asn.gz
gunzip linux64.tbl2asn.gz 
mv linux64.tbl2asn tbl2asn
chmod +x tbl2asn
cp tbl2asn /usr/local/bin
</code></pre></div>
<p>Install <a href="http://www.biostars.org/p/63816/">GNU Parallel</a>:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O http://ftp.gnu.org/gnu/parallel/parallel-20130822.tar.bz2
tar xjvf parallel-20130822.tar.bz2
cd parallel-20130822/
ls
./configure &amp;&amp; make &amp;&amp; make install
</code></pre></div>
<p>Install <a href="http://infernal.janelia.org/">Infernal</a>:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd /mnt
curl -O http://selab.janelia.org/software/infernal/infernal-1.1rc4.tar.gz
tar xzf infernal-1.1rc4.tar.gz 
cd infernal-1.1rc4/
ls
./configure &amp;&amp; make &amp;&amp; make install
</code></pre></div>
<h5 id="running-prokka-at-the-command-line">Running Prokka at the command line</h5>

<p>First we&#39;ll use the khmer tool to remove all the sequences with &#39;N&#39;s in them (since prodigal fails if there are too many, and prokka uses prodigal):</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">python /usr/local/share/khmer/sandbox/remove-N.py final-assembly.fa metagenome.fa
</code></pre></div>
<p>Now, run Prokka::</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">/mnt/prokka-1.7/bin/prokka metagenome.fa --outdir annotation --prefix assembly --metagenome
</code></pre></div>
<p>There will be a bunch of files in the directory <code>annotation</code>.  Probably the most interesting is <code>assembly/assembly.faa</code>, which will contain a set of
annotated protein sequences derived from the metagenome.</p>

<p>Let&#39;s look at some of the files now!</p>

<h4 id="3.-assessing-assembly-quality-with-quast"><strong>3.</strong> Assessing Assembly Quality with QUAST</h4>

<p>We&#39;re going to try to get an idea of what our assembly looks like by using the <a href="http://bioinf.spbau.ru/quast">QUAST (Quality Assessment Tool for Genome Assemblies)</a>.</p>

<p>First we need to have some tools installed:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">sudo apt-get install python-matplotlib
</code></pre></div><div class="highlight"><pre><code class="language-text" data-lang="text">cd ~
mkdir PROGRAMS
cd PROGRAMS
wget https://downloads.sourceforge.net/project/quast/quast-2.3.tar.gz
tar xzvf quast-2.3.tar.gz
cd quast-2.3
</code></pre></div>
<p>The program is installed here, but now we want to run a test to make sure:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">python metaquast.py --test
</code></pre></div>
<p>A bunch of code will pass by, at the end you will see:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">MetaQUAST finished.
Log saved to /home/ubuntu/PROGRAMS/quast-2.3/quast_test_output/metaquast.log

Finished: 2014-08-19 15:26:24
Elapsed time: 0:00:54.979271
Total NOTICEs: 12; WARNINGs: 0; non-fatal ERRORs: 0

Thank you for using QUAST!
</code></pre></div>
<p>...and it passed! Now we don&#39;t have to do anything else to have a working QUAST.</p>

<p>We want to now do an assessment of our scaffolds:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">python metaquast.py -o folder ~/assembly/scaffolds.fasta
</code></pre></div>
<p>Now we have a <code>folder</code> with our results (it&#39;s pretty quick, which is a big plus after the rest of the tools we&#39;ve used).</p>

<p>Let&#39;s look at the files:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">cd folder/quast_output
ls -la
</code></pre></div>
<p>Now you will see:</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">total 96
drwxrwxr-x 4 ubuntu ubuntu  4096 Aug 19 15:37 .
drwxrwxr-x 4 ubuntu ubuntu  4096 Aug 19 15:37 ..
drwxrwxr-x 2 ubuntu ubuntu  4096 Aug 19 15:37 basic_stats
-rw-rw-r-- 1 ubuntu ubuntu  2105 Aug 19 15:37 quast.log
-rw-rw-r-- 1 ubuntu ubuntu 23161 Aug 19 15:37 report.html
drwxrwxr-x 6 ubuntu ubuntu  4096 Aug 19 15:37 report_html_aux
-rw-rw-r-- 1 ubuntu ubuntu 25569 Aug 19 15:37 report.pdf
-rw-rw-r-- 1 ubuntu ubuntu   799 Aug 19 15:37 report.tex
-rw-rw-r-- 1 ubuntu ubuntu   251 Aug 19 15:37 report.tsv
-rw-rw-r-- 1 ubuntu ubuntu   677 Aug 19 15:37 report.txt
-rw-rw-r-- 1 ubuntu ubuntu   704 Aug 19 15:37 transposed_report.tex
-rw-rw-r-- 1 ubuntu ubuntu   251 Aug 19 15:37 transposed_report.tsv
-rw-rw-r-- 1 ubuntu ubuntu   547 Aug 19 15:37 transposed_report.txt
</code></pre></div>
<p>We can now look at the HTML report or the PDF report.  You can <code>scp</code> these files, or use a method to move files to your computer.</p>

<p>Let&#39;s look at the assembly stats.  </p>


<div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'edamame'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
    

                    </div>

                
            </div>


            <div class=row-fluid>
                <div id=footer class=span12>
                    Documentation for <a href="http://edamame-course.org">EDAMAME</a>.

                </div>
            </div>
        </div>

        <script>
            function orderNav() {
                var list,
                    section,
                    header,
                    sections = [],
                    lists = {},
                    headers = {};

                var navUl = document.querySelectorAll('#navigation ul')[0],
                    navLis = document.querySelectorAll('#navigation ul li');

                if (!navUl) return;

                for (var i = 0; i < navLis.length; i++) {
                    var order, li = navLis[i];

                    if (li.classList.contains('nav-header')) {
                        section = li.textContent || li.innerText;
                        sections.push(section);
                        headers[section] = li;
                        continue;
                    }

                    if (!lists[section]) {
                        lists[section] = [];
                    }

                    order = parseFloat(li.getAttribute('data-order'))
                    lists[section].push([order, li]);
                }

                for (var i = 0; i < sections.length; i++) {
                    section = sections[i];
                    list = lists[section].sort(function(a, b) {
                        return a[0] - b[0];
                    });

                    if (header = headers[section]) {
                        navUl.appendChild(header);
                    }
                    for (var j = 0; j < list.length; j++) {
                        navUl.appendChild(list[j][1]);
                    }
                }
            }

            if (document.querySelectorAll) orderNav();
        </script>
    </body>
</html>
