<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="viewport" content="width=device-width">

        <title>EDAMAME : Moving data from one program to another</title>
        <meta name="description" content="course materials & tutorials">

        <link href="//netdna.bootstrapcdn.com/twitter-bootstrap/2.3.2/css/bootstrap-combined.no-icons.min.css" rel="stylesheet">
        <link rel="stylesheet" href="/css/syntax.css">
        <link rel="stylesheet" href="/css/main.css">
    </head>
    <body>

        <div class="container">
            <div class=row-fluid>
                <div id=header class=span12>
                    <h4><a class=brand href="http://edamame-course.org">EDAMAME</a>
    <small>course materials & tutorials</small>
</h4>

                </div>
            </div>

            <div class=row-fluid>
                
                
                    <div id=navigation class=span3>
                        
<ul class="nav nav-list">
    <li class=nav-header>HOME</li>
    <li><a href="index.html">EDAMAME course materials</a></li>
    <li class=divider></li>
    <li class=nav-header>Before We Begin</li>
    <li><a href="https://edamame-course.github.io/docs/general_set_up_guidelines.html">Getting Set Up For The Course</a></li>
    <li><a href="https://edamame-course.github.io/docs/intro_to_ec2_instance.html">All Users: Setting up an EC2 instance</a></li>
    <li><a href="https://edamame-course.github.io/docs/mobaxterm.html">Windows Users: Connecting to your AWS EC2 instance</a></li>
    <li><a href="https://edamame-course.github.io/docs/mac_os_linux_connect_to_ec2.html">Mac and Linux Users: Connecting to your AWS EC2 instance</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/macqiime_installation.html">Mac Users: Installing MacQIIME</a></li>
    <li><a href="https://edamame-course.github.io/docs/QIIME_VB_for_Windows.html">Windows Users: Installing a Virtual Box to run QIIME on your own Windows computer</a></li>
    <li class=divider></li>
    <li class=nav-header>Wednesday, August 13th</li>
    <li><a href="https://edamame-course.github.io/docs/august_13.html">Wednesday, August 13th Overview</a></li>
    <li><a href="https://edamame-course.github.io/docs/the_shell.html">An Introduction To Using The Shell</a></li>
    <li><a href="https://edamame-course.github.io/docs/intro_qiime_tutorial.html">Introduction to Data Analysis with QIIME</a></li>
    <li class=divider></li>
    <li class=nav-header>Thursday, August 14th</li>
    <li><a href="https://edamame-course.github.io/docs/august_14.html">Thursday, August 14th Overview</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-08-14-AM_Ashley_Lecture2.pdf?raw=true">Alpha Diversity Lecture</a></li>
    <li><a href="https://edamame-course.github.io/docs/qiime_2_tutorial.html">QIIME Tutorial Part Two</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-08-14-PM_Ashley_Lecture3.pdf?raw=true">Beta Diversity Lecture</a></li>
    <li><a href="https://edamame-course.github.io/docs/qiime_3.html">QIIME Tutorial Part Three</a></li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-edamame-assembly_TBrown.pdf?raw=true">Titus Brown's Intro to Metagenomics Lecture</a></li>
    <li class=divider></li>
    <li class=nav-header>Friday, August 15th</li>
    <li><a href="https://edamame-course.github.io/docs/august_15.html">Friday, August 15th Overview</a></li>
    <li><a href="https://edamame-course.github.io/docs/intro_to_ec2_instance.html">Introduction to using Amazon EC2</a></li>
    <li><a href="http://pschloss.github.io/2014_08_15-mothur/#1">Pat Schloss's Presentation</a></li>
    <li><a href="http://www.mothur.org/wiki/MiSeq_SOP">mothur to the max</a></li>
    <li class=divider></li>
    <li class=nav-header>Saturday, August 16th</li>
    <li><a href="https://edamame-course.github.io/docs/august_15.html">Saturday, August 16th Overview</a></li>
    <li><a href="">Introduction to Metagenomics</a></li>
    <li><a href="">Using BLAST at the command line</a></li>
    <li class=divider></li>
    <li class=nav-header>Sunday, August 17th</li>
    <li>Brunch with Jim Tiedje!</li>
    <li><a href="">Metagenome Assembly</a></li>
    <li class=divider></li>
    <li class=nav-header>Monday, August 18th</li>
    <li><a href="https://github.com/edamame-course/docs/august_18.html">Monday, August 18th Overview</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-18-AM_Ashley_Lecture4.pdf?raw=true">Hypothesis testing, identifying gradients, and linking environmental and community data</a></li>
    <li><a href="https://github.com/edamame-course/docs/tree/gh-pages/extra/Jones_R_Tutorial_Files">Using R for community ecology analyses</a></li>
    <li><a href=""></a></li>
    <li class=divider></li>
    <li class=nav-header>Tuesday, August 19th</li>
    <li><a href="">Jim Cole - Guest Lecture</a></li>
    <li><a href="http://adina-howe.readthedocs.org/en/latest/">Adina Howe: Tutorial on working with NCBI and MG-RAST and their APIs</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-19-Docherty_NEON.pdf?raw=true">Kathryn Docherty - Guest Lecture</a></li>
    <li><a href="https://edamame-course.github.io/docs/metagenome_assembly_and_annotation.html">Josh: Metagenome Assembly, Annotation, and Quality Assessment</a></li>
    <li><a href="https://edamame-course.github.io/docs/using_dropbox_on_your_ec2_machine.html">Josh: Installing Dropbox on any computer from the command line</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/_posts/2014-08-19_database_questions.md">Self Guided Tour of Databases</a></li>
    <li><a href="">Josh speaks about the wild and woolly world of other marker regions!</a></li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-19-Peralta.pdf?raw=true"></a>Guest lecture from Ariane Peralta on Human-impacted ecosystems</li>
    <li><a href="https://github.com/edamame-course/docs/blob/gh-pages/extra/Presentations/2014-08-20-metagenomic_visualization.pdf?raw=true">Tracy gives a tour of data visualization!</a></li>
    <li class=divider></li>
    <li class=nav-header>Wednesday, August 20th</li>
    <li><a href="https://edamame-course.github.io/docs/extra/Presentations/2014-08-20_Cluster_Computing_EDAMAME.pdf?raw=true">Guest Lecture from Dirk Colbry - Using a Computer Cluster</a></li>
    <li>Course Wrap-Up and Evaluation</li>
</ul>

                    </div>

                    <div id=content class=span9>
                        <div class=page-header>
    <h2>Moving data from one program to another
        
    </h2>
</div>

<h3 id="1.-moving-data-from-qiime:-converting-the-.biom-table"><strong>1.</strong>  Moving data from QIIME:  converting the .biom table</h3>

<p>This command changes frequently, as the biom format is a work in progress.  Use <code>biom convert -h</code> to find the most up-to-date arguments and options; the web page is not updated as frequently as the help file.</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">biom convert -b -i Schloss_otu_table_even2212.biom -o Schloss_otu_table_even2212.txt --table-type otu  --header-key taxonomy --output-metadata-id &quot;ConsensusLineage&quot;
</code></pre></div>
<p>Here, we use the argument <code>-b</code> to specify that we are converting a biom-formatted table to a &quot;classic&quot; otu table.  We specify input and output files with <code>-i</code> and <code>-o</code>, as always.  We also provide the <code>--table-type</code> arugment to &quot;otu.&quot;  Finally, we specify that we want our OTU table to have the taxonomy assigned to each OTU as the last column in the table (<code>--header-key</code> set to &quot;taxonomy&quot;) and that the name of this column, <code>--output-metadata-id</code>, will be the &quot;ConsensusLineage&quot;</p>

<h3 id="2.-qiime-to-r-:-resemblance-tables"><strong>2.</strong>  QIIME to R  : Resemblance tables</h3>

<p>For importing resemblance tables
    1.  read.table(header=TRUE, row.names =1; sep =“\t”)
    2.  as.dist()</p>

<h3 id="3.-writing-out-(exporting)-resemblance-mtables-from-r"><strong>3.</strong> Writing out (exporting) resemblance mtables from R</h3>

<p>It is advisable to save resemblance matrices/OTU to a working directory so that we don&#39;t have to re-calculate them every time we continue analyses, or if we want to use these matrices in a different program.</p>

<p>Remember that resemblance matrices are actually very long vectors.  R needs to think of these as tables so that it can write them out with the correct delimitations.  Thus, we take the opposite approach as we did when we read in the table. We use the &quot;as.matrix()&quot; command to convert the vector to a table.  Below is an example.</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">braycurtis=as.matrix(braycurtis.d)
</code></pre></div>
<p>Then, we use the the &quot;write.table()&quot; command to export the data.  In this command, the first argument is the object that we want to export, the second argument  (given in quotations) is what we want the name of the exported file to be - don&#39;t forget the extension, we set row.names and col.names to TRUE because we have both, we specify that we want the exported file to be tab-delimited by setting the sep=&quot;\t&quot; argument, and, finally, we specify quote=FALSE, as otherwise R will put every value in the exported table in quotes, which is annoying.  Use the same commands with the Sorenson matrix.</p>
<div class="highlight"><pre><code class="language-text" data-lang="text">write.table(braycurtis, &quot;BrayCurtis_even861.txt&quot;,row.names=TRUE, col.names=TRUE,sep=&quot;\t&quot;, quote=FALSE)
</code></pre></div><div class="highlight"><pre><code class="language-text" data-lang="text">sorenson=as.matrix(sorenson.d)
</code></pre></div><div class="highlight"><pre><code class="language-text" data-lang="text">write.table(sorenson, &quot;Sorenson_even861.txt&quot;,row.names=TRUE, col.names=TRUE,sep=&quot;\t&quot;, quote=FALSE)
</code></pre></div>
<p>If you open these resemblance matrices in Excel, you will notice another formatting issue.  You must insert a top leftmost cell so that the sample IDs (column names) are shifted one over.  This may seem confusing, but inspect one of the exported files in Excel and everything will be illuminated.</p>

<hr>

<hr>


<div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'edamame'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
    

                    </div>

                
            </div>


            <div class=row-fluid>
                <div id=footer class=span12>
                    Documentation for <a href="http://edamame-course.org">EDAMAME</a>.

                </div>
            </div>
        </div>

        <script>
            function orderNav() {
                var list,
                    section,
                    header,
                    sections = [],
                    lists = {},
                    headers = {};

                var navUl = document.querySelectorAll('#navigation ul')[0],
                    navLis = document.querySelectorAll('#navigation ul li');

                if (!navUl) return;

                for (var i = 0; i < navLis.length; i++) {
                    var order, li = navLis[i];

                    if (li.classList.contains('nav-header')) {
                        section = li.textContent || li.innerText;
                        sections.push(section);
                        headers[section] = li;
                        continue;
                    }

                    if (!lists[section]) {
                        lists[section] = [];
                    }

                    order = parseFloat(li.getAttribute('data-order'))
                    lists[section].push([order, li]);
                }

                for (var i = 0; i < sections.length; i++) {
                    section = sections[i];
                    list = lists[section].sort(function(a, b) {
                        return a[0] - b[0];
                    });

                    if (header = headers[section]) {
                        navUl.appendChild(header);
                    }
                    for (var j = 0; j < list.length; j++) {
                        navUl.appendChild(list[j][1]);
                    }
                }
            }

            if (document.querySelectorAll) orderNav();
        </script>
    </body>
</html>
