<h3 id="handouts-of-workflow-charts-are-available-for-the-qiime-workflow-discussed-in-these-tutorials">Handouts of workflow charts are available for the QIIME workflow discussed in these tutorials:</h3>
<ul>
  <li><a href="https://github.com/edamame-course/docs/tree/gh-pages/extra/Handouts/QIIMEFlowChart_IlluminaPairedEnds_13aug2014.pdf?raw=true">Paired-End Illumina</a></li>
  <li><a href="https://github.com/edamame-course/docs/tree/gh-pages/extra/Handouts/QIIMEFlowChart_454_13aug2014.pdf?raw=true">454</a></li>
</ul>

<h2 id="getting-started">Getting started</h2>

<p>Make a new directory <code>mkdir</code> in which to put all of your QIIME-related analyses for today and tomorrow, and then ‘cd’ to move into that directory.  Execute all commands from within this directory.</p>

<p><code>
mkdir QIIMETutorial
</code></p>

<p><code>
cd QIIMETutorial
</code></p>

<h2 id="assembling-illumina-paired-end-sequences">Assembling Illumina paired-end sequences</h2>

<h3 id="download-schloss-mouse-data">1.1  Download Schloss mouse data</h3>
<p>These data are 16S rRRNA V4 amplicons sequenced with MiSeq technology:  </p>

<p><em>If you are a Mac user</em>, about half-way down the page, click on <a href="http://www.mothur.org/wiki/MiSeq_SOP">Example data from Schloss lab - http://www.mothur.org/wiki/MiSeq_SOP</a>.  </p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_01.jpg" alt="img1" />.</p>

<p>Unzip (the directory with the data will be called MiSeq_SOP) and move it into the “QIIMETutorial” directory.  </p>

<p><em>If you are on the EC2 or the QIIME Virtual Box</em>, download the Schloss data from the terminal, using <code>wget</code> :  </p>

<p><code>
wget http://www.mothur.org/w/images/d/d6/MiSeqSOPData.zip  
</code></p>

<p><code>
unzip MiSeqSOPData.zip  
</code></p>

<p>This will make the directory “MISeq_SOP.”  Move this into the QIIMETutorial directory.</p>

<h3 id="use-mkdir-to-create-a-new-directory-called-pandaseqmergedreads">1.2  Use <code>mkdir</code> to create a new directory called “pandaseq_merged_reads”</h3>

<p><code>
mkdir pandaseq_merged_reads
</code></p>

<h3 id="join-paired-end-illumina-reads-with-pandaseq">1.3  Join paired end Illumina reads with PANDAseq**</h3>
<p><code>
pandaseq -f MiSeq_SOP/F3D0_S188_L001_R1_001.fastq -r MiSeq_SOP/F3D0_S188_L001_R2_001.fastq -w pandaseq_merged_reads/F3D0_S188.fasta -g pandaseq_merged_reads/F3D0_S188.log -L 255 -t 0.90
</code>
Let’s look carefully at the anatomy of this command.</p>

<ul>
  <li><code>pandaseq</code> calls the package of pandaseq scripts.</li>
  <li><code>-f MiSeq_SOP/F3D0_S188_L001_R1_001.fastq</code> tells the script where to find the forward read.</li>
  <li><code>-r</code> tells the script where to find its matching reverse read.</li>
  <li><code>-w pandaseq_merged_reads/F3D0_S188.fasta</code> directs the script to make a new fasta file of the assembled reads and to put it in the “pandaseq_merged_reads” directory.</li>
  <li><code>-g pandaseq_merged_reads/F3D0_S188.log</code> Selects an option of creating a log file.</li>
  <li><code>-L</code> specifies the maximum length of the assembled reads, which, in truth, should be 251 bp. This is a very important option to specify, otherwise PANDAseq will assemble a lot of crazy-long sequences.</li>
</ul>

<p>All of the above information, and more options, are fully described in the <a href="http://neufeldserver.uwaterloo.ca/~apmasell/pandaseq_man1.html">PANDAseq Manual.</a>.  The log file includes details as to how well the merging went.</p>

<h3 id="sanity-check-1-and-file-inspection">1.4  Sanity check #1 and file inspection.</h3>

<p>There are some questions you may be having: What does pandaseq return?  Are there primers/barcodes on the assembled reads?  Are these automatically trimmed?</p>

<p>It turns out, that PANDAseq, by default, removes primers and barcodes (There are also options to keep primers, please see the manual link above).  Given that we used the default pandaseq options, how do we check to make sure that what we expect to happen actually did happen?</p>

<p>We know the V4 forward primer sequence that the Schloss lab used because these sequences are provided in Kozich et al. 2013.</p>

<p>Here is the V4 primer sequence : GTCCAGCMGCCGCGGTAA</p>

<p>We can search for that sequence in the assembled sequence file, using the <code>grep</code> function.  </p>

<p><code>
cd pandaseq_merged_reads
</code></p>

<p><code>
head F3D0_S188.fasta
</code></p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_02.jpg" alt="img2" />  </p>

<p><code>
grep GTCCAGCMGCCGCGGTAA F3D0_S188.fasta
</code></p>

<p>When you execute the above command, the terminal does not return anything.  This means that the primer sequence was not found in the file, suggesting that PANDAseq did, in fact, trim them.</p>

<p>We can double check our sanity by using a positive control.  Let’s use <code>grep</code> to find a character string that we know is there, for instance the “M00967” string identifying the first sequence.</p>

<p><code>
grep M00967 F3D0_S188.fasta
</code></p>

<p>Whoa!  That is hard to read all of those lines. Let’s put the results into a list by appending <code>&gt; list.txt</code> to the command.  The “&gt;” symbol means to output the results to a new file, which is specified next.  </p>

<p><code>
grep M00967 F3D0_S188.fasta &gt; list.txt
</code></p>

<p>This creates a new file called “list.txt”, in which all instances of the character string “M00967” are provided.  Let’s look at the head.</p>

<p><code>
head list.txt
</code></p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_03.jpg" alt="img3" />  </p>

<p>Our positive control worked, and we should be convinced and joyous that we executed <code>grep</code> correctly AND that the primers were trimmed by PANDAseq.  We can now remove the list file.</p>

<p><code>
rm list.txt
</code></p>

<h3 id="automate-paired-end-merging-with-a-shell-script">1.5  Automate paired-end merging with a shell script.</h3>

<p>We would have to execute an iteration of the PANDAseq command for every pair of reads that need to be assembled. This could take a long time.  So, we’ll use a shell script to automate the task.  </p>

<p>Download this <a href="https://github.com/edamame-course/docs/raw/gh-pages/misc/QIIMETutorial_Misc/SchlossSampleNames.txt">list</a> (<em>VB/EC2 users</em>, use <code>wget</code>) of all the unique sample names and move it to your QIIMETutorial directory.  </p>

<p>Then, download this shell <a href="https://github.com/edamame-course/docs/raw/gh-pages/misc/QIIMETutorial_Misc/pandaseq_merge.sh">script</a> (<em>VB/EC2 users</em>, use <code>wget</code>) and move it to your QIIMETutorial directory.  </p>

<p>Change permissions on the script</p>

<p><code>
chmod +x pandaseq_merge.sh
</code></p>

<p>Execute the script from the QIIMETutorial Directory.</p>

<p><code>
./pandaseq_merge.sh
</code></p>

<h3 id="sanity-check-2">1.6  Sanity check #2.</h3>

<p>How many files were we expecting from the assembly?  There were 19 pairs to be assembled, and we are generating one assembled fasta and one log for each.  Thus, the pandaseq_merged_reads directory should contain 38 files.  We use the <code>wc</code> command to check the number of files in the directory.</p>

<p><code>
ls -1 pandaseq_merged_reads | wc -l
</code></p>

<p>The terminal should return the number “38.”  Congratulations, you lucky duck! You’ve assembled paired-end reads!  </p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_04.jpg" alt="img4" />  </p>

<h2 id="moving-assembled-reads-into-the-qiime-environment">Moving assembled reads into the QIIME environment</h2>

<p>While working through the tutorial, open your web browser and navigate to this <a href="http://www.qiime.org/scripts/index.html">page</a>. It provides an index of qiime scripts and options.  We will be using the default for most of the time, but for each script, it is useful to open its documentation and assess the alternative options.</p>

<h3 id="understanding-the-qiime-mapping-file">2.1  Understanding the QIIME mapping file.</h3>
<p>QIIME requires a <a href="http://qiime.org/documentation/file_formats.html">mapping file</a> for most analyses.  This file is important because it links the sample IDs with their metadata (and, with their primers/barcodes if using QIIME for quality-control). Because we are super-amazing, we’ve already created a mapping file for the Schloss data for you.  <a href="https://github.com/edamame-course/docs/raw/gh-pages/misc/QIIMETutorial_Misc/Schloss_Map.txt">Download it</a> (<em>VB/EC2</em> users, use <code>wget</code>), and move it to your QIIMETutorial directory.</p>

<p>Let’s spend few moments getting to know the mapping file:</p>

<p><code>
more Schloss_Map.txt
</code></p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_05.jpg" alt="img5" />  </p>

<p>A clear and comprehensive mapping file should contain all of the information that will be used in downstream analyses.  The mapping file includes both categorical (qualitative) and numeric (quantitative) contextual information about a sample. This could include, for example, information about the subject (sex, weight), the experimental treatment, time or spatial location, and all other measured variables (e.g., pH, oxygen, glucose levels). Creating a clear mapping file will provide direction as to appropriate analyses needed to test hypotheses.  Basically, all information for all anticipated analyses should be in the mapping file.</p>

<p><em>Hint</em>:  Mapping files are also a great way to organize all of the data for posterity in the research group.  New lab members interested in repeating the analysis should have all of the required information in the mapping file.  PIs should ask their students to curate and deposit both mapping files and raw data files.</p>

<p>Guidelines for formatting map files:
  - Mapping files should be tab-delimited
  - The first column must be “#SampleIDs” (commented out using the <code>#</code>).
  -  SampleIDs are VERY IMPORTANT. Choose wisely! Ideally, a user who did not design the experiment should be able to distiguishes the samples easily, as is the case with the Schloss data. SampleIDs must be alphanumeric characters or periods.  They cannot have underscores.
  - The last column must be “Description”.
  - There can be as many in-between columns of contextual data as needed.
  - If you plan to use QIIME for quality control (which we do not need because the PANDAseq merger included QC), the BarcodeSequence and LinkerPrimer sequence columns are also needed, as the second and third columns, respectively.
  - Excel can cause formatting heartache.  See more details <a href="misc/QIIMETutorial_Misc/MapFormatExcelHeartAche.md">here</a>.</p>

<h3 id="call-qiime">2.2  Call QIIME</h3>

<p>For Mac users, to enter the QIIME environment in all of its glory, use <code>macqiime</code>.</p>

<p>A good command to know is:</p>

<p><code>
print_qiime_config.py
</code></p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_09.jpg" alt="img9" />  </p>

<p>This will give you really important information about versions of software, etc.  You will need this info</p>

<h3 id="merging-assembled-reads-into-the-one-big-ol-data-file">2.3  Merging assembled reads into the one big ol’ data file.</h3>

<p>QIIME expects all of the data to be in one file, and, currently, we have one separate fastq file for each assembled read.  We will add labels to each sample and merge into one fasta using the <code>add_qiime_labels.py</code> script. Documentation is <a href="http://qiime.org/scripts/add_qiime_labels.html">here</a>.</p>

<p><code>
add_qiime_labels.py -i pandaseq_merged_reads/ -m Schloss_Map.txt -c InputFileName -n 1 -o combined_fasta
</code></p>

<p>This script creates a new directory called “combined_fasta.”  Use <code>cd</code> and <code>ls</code> to navigate to that directory and examine the files.  Inspect the new file “combined_seqs.fna.”</p>

<p><code>
head combined_seqs.fna
</code></p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_06.jpg" alt="img6" />  </p>

<p>Observe that QIIME has added the SampleIDs from the mapping file to the start of each sequence.  This allows QIIME to quickly link each sequence to its sampleID and metadata.</p>

<p>While we are inspecting the combined_seqs.fna file, let’s confirm how many sequences we have in the dataset.</p>

<p><code>
count_seqs.py -i combined_seqs.fna
</code></p>

<p>This is a nice QIIME command to call frequently, because it provides the total number of sequences in a file, as well as some information about the lengths of those sequences.  But, suppose we wanted to know more than the median/mean of these sequences?</p>

<p>Another trick with QIIME is that you can call all the mothur commands within the QIIME environment, which is very handy.  mothur offers a very useful command called <code>summary.seqs</code>, which operates on a fasta/fna file to give summary statistics about its contents.  We will cover mothur in all its glory later, but for now, execute the command:</p>

<p><code>
mothur
</code></p>

<p><code>
summary.seqs(fasta=combined_seqs.fna)
</code></p>

<p>Note that both summary.seqs and count_seqs.py have returned the same total number of seqs in the .fna file.  Use the following command to quit the mothur environment and return to QIIME.  </p>

<p><img src="img/QIIMETutorial1_IMG/summary_seqs.jpg" alt="img" /></p>

<p><code>
quit()
</code></p>

<h5 id="optional-step---chimera-checking-with-usearch-before-picking-otus--usearch-is-an-add-on-to-macqiime-extra-installation-steps-required">2.3.2  Optional step* :  chimera checking with USEARCH before picking OTUs.  USEARCH is an add-on to MacQIIME (extra installation steps required).</h5>

<h3 id="picking-operational-taxonomic-units-otus">2.4  Picking Operational Taxonomic Units, OTUs.</h3>
<p>Picking OTUs is sometimes called “clustering,” as sequences with some threshold of identity are “clustered” together to into an OTU.</p>

<p><em>Important decision</em>: Should I use a de-novo method of picking OTUs or a reference-based method, or some combination? (<a href="http://www.mendeley.com/catalog/interpreting-16s-metagenomic-data-without-clustering-achieve-subotu-resolution/">Or not at all?</a>). The answer to this will depend, in part, on what is known about the community a priori.  For instance, a human or mouse gut bacterial community will have lots of representatives in well-curated 16S databases, simply because these communities are relatively well-studied.  Therefore, a reference-based method may be preferred.  The limitation is that any taxa that are unknown or previously unidentified will be omitted from the community.  As another example, a community from a lesser-known environment, like Mars or a cave, or a community from a relatively less-explored environment would have fewer representative taxa in even the best databases.  Therefore, one would miss a lot of taxa if using a reference-based method.  The third option is to use a reference database but to set aside any sequences that do not have good matches to that database, and then to cluster these de novo.</p>

<p>We use the <code>pick_otus.py</code> script in QIIME for this step.  Documentation is <a href="http://qiime.org/scripts/pick_otus.html?highlight=pick_otus">here</a>.
The default QIIME 1.8.0 method for OTU picking is uclust (de novo, but there is a reference-based alternative, see below), but we will use the CD-HIT algorithm (de novo).  However, we encourage you to explore different OTU clustering algorithms to understand how they perform.  They are not created equal.  Honestly, we are using CD-HIT here because because it is fast.</p>

<p>Make sure you are in the QIIMETutorial directory to start.  This will take a few (&lt;10ish) minutes.</p>

<p><code>
pick_otus.py -i combined_fasta/combined_seqs.fna -m cdhit -o cdhit_picked_otus/ -s 0.97 -n 100
</code></p>

<p>In the above script:
  - We tell QIIME to look in the “combined_fasta” directory for the input file <code>-i</code>, “combined_seqs.fna”.
  - We chose the clustering method CD-HIT <code>-m</code>
  - We defined an output file “cdhit_picked_otus” <code>-o</code>.  Names of output files are important, because there are many options for each analysis. Using the algorithm choice in the directory name is key for comparing the output of multiple algothims (for instance, if you wanted to compare how picking OTUs with CD-HIT and with uclust may influence your results.)
  - We define OTUs at 97% sequence identity <code>-s 0.97</code>
  - We opt for a pre-filtering step, unique to CD-HIT <code>-n</code> = 100.</p>

<p>Inspect the log and the resulting combined_seqs_otus.txt file, using <code>head</code>.  You should see an OTU ID (yellow box), starting at “0” the the left most column.  After that number, there is a list of Sequence IDs that have been clustered into that OTU ID.  The first part of the sequence ID is the SampleID from which it came (green box), and the second part is the sequence number within that sample (purple box).  </p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_07.jpg" alt="img7" /></p>

<p>From the head of the combined_seqs_otus.txt file, we can see that OTU 0 has many sequence associated with it, including sequence 9757 from from sample F3D8.S196. We also see that OTU 3 only has one sequence associated with it. The log file has goodies about the algorithm and options chosen.  Keep this (and all) log file, because when you are writing the paper you may not remember what version of which clustering algorithm you used.</p>

<h3 id="pick-a-representative-sequence-from-each-otu">2.5  Pick a representative sequence from each OTU.</h3>

<p>Representative sequences are those that will be aligned and used to build a tree.  They are selected as the one sequence, out of its whole OTU cluster, that will “define” its OTUs. As you can imagine, understanding how these “rep seqs” are chosen is very important.  Here, we will use the default method (the first sequence listed in the OTU cluster) of QIIME’s <code>pick_rep_set.py</code> script; documentation <a href="http://qiime.org/scripts/pick_rep_set.html">here</a>.</p>

<p><code>
mkdir cdhit_rep_seqs/
</code></p>

<p><code>
pick_rep_set.py -i cdhit_picked_otus/combined_seqs_otus.txt -f combined_fasta/combined_seqs.fna -o cdhit_rep_seqs/cdhit_rep_seqs.fasta -l cdhit_rep_seqs/cdhit_rep_seqs.log
</code></p>

<p>As before, we specify the input files (the script needs the OTU clusters and the raw sequence file as input), and then we additionally specified the a new directory for the results.
Inspect the head of the new fasta file, cdhit_rep_seqs.fasta.</p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_08.jpg" alt="img8" /></p>

<p>As before, we see the OTU ID given first (consecutively, starting with 0), and then the sequence ID of the representative sequence, and then the full nucleotide information for the sequence. Notice that for OTU 0, which only had one sequence in its “cluster”, is defined by that one sequence.  Don’t be shy - go ahead and compare it to the combined_seqs_otus.txt file of OTU clusters.</p>

<p>Take a gander at the log file, as well.  </p>

<h3 id="align-representative-sequences">2.6 Align representative sequences.</h3>

<p>Navigate back to the QIIMETutorial directory. We will align our representative sequences using PyNAST, which uses a “gold” reference template for the alignment.  QIIME uses a “gold” pre-aligned template made from the greengenes database.  The default alignment to the template is minimum 75% sequence identity and minimum length 150. The default minimum length is not great for short reads like we have, so we will be more generous and change the default. What should we change it to?</p>

<p><code>
count_seqs.py -i cdhit_rep_seqs/cdhit_rep_seqs.fasta
</code></p>

<p>Given that our average assembled read length is ~252 bp, let’s decide that at least 100 bp must align.  We will have the <code>-e</code> option to 100. The alignment will take a few minutes.  Documentation for <code>align_seqs.py</code> is <a href="http://qiime.org/scripts/align_seqs.html">here</a>.</p>

<p><code>
align_seqs.py -i cdhit_rep_seqs/cdhit_rep_seqs.fasta -o pynast_aligned/ -e 100
</code></p>

<p>Navigate into the pynast_aligned directory.  There are three files waiting there:  one file of sequences that failed to align, one of sequences that did align, and a log file.  Inspect each.</p>

<p><code>
count_seqs.py -i cdhit_rep_seqs_failures.fasta
</code></p>

<p><code>
count_seqs.py -i cdhit_rep_seqs_aligned.fasta
</code></p>

<p>We see that there were ~3 rep. sequences that failed to align, and approximately 682 that did.  (Also, notice what short-read alignments generally look like…not amazing).</p>

<p><em>Sanity check?</em>  If you like, <a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastSearch&amp;BLAST_SPEC=MicrobialGenomes">BLAST</a> the top sequence that failed to align to convince yourself that it is, indeed, a pitiful failure.</p>

<p>If, in the future, you ever have a large proportion of rep seqs that fail to align, it could be due to:
  *  Hooray! These are novel organisms! (But, think about the habitat before jumping to conclusions. As lucky as we would be to discover new species in the mouse gut, this is unlikely.)
  *  The alignment parameters were too stringent for short reads, causing “real” sequences to fail alignment.
  * The paired-end merger algorithm (e.g., pandaseq) did not do a perfect job, and concatenated ends that do not belong together.
  * Some combination of the above, as well as some other scenarios.</p>

<p>We will filter out these failed-to-align sequences (really, the removing the entire OTU cluster that they represent) from the dataset after we make the OTU table.  In the meantime, let’s create a text file of all the names of the rep. sequence OTU IDs that we want to remove.  We only have three failures, so we easily could do it by hand.  What if we had more?  Here’s how to automate the generation of the “cdhit_rep_seqs_failures_names.txt” file using the <code>grep</code> command. We will not go into details, but general grep help is <a href="http://unixhelp.ed.ac.uk/CGI/man-cgi?grep">here</a>. Navigate back into the QIIMETutorial directory to run the grep command.</p>

<p><code>
grep -o -E "^&gt;\w+" pynast_aligned/cdhit_rep_seqs_failures.fasta | tr -d "&gt;" &gt; pynast_aligned/cdhit_rep_seqs_failures_names.txt
</code></p>

<p>Congratulations!  You just had the QIIME of Your Life!</p>

<p><img src="https://github.com/edamame-course/docs/raw/gh-pages/img/QIIMETutorial1_IMG/IMG_10.jpg" alt="img10" />  </p>

<h2 id="where-to-find-qiime-resources-and-help">Where to find QIIME resources and help</h2>
<ul>
  <li><a href="qiime.org">QIIME</a> offers a suite of developer-designed <a href="http://www.qiime.org/tutorials/tutorial.html">tutorials</a>.</li>
  <li><a href="http://www.qiime.org/scripts/index.html">Documentation</a> for all QIIME scripts.</li>
  <li>There is a very active <a href="https://groups.google.com/forum/#!forum/qiime-forum">QIIME Forum</a> on Google Groups.  This is a great place to troubleshoot problems, responses often are returned in a few hours!</li>
  <li>The <a href="http://qiime.wordpress.com/">QIIME Blog</a> provides updates like bug fixes, new features, and new releases.</li>
  <li>QIIME development is on <a href="https://github.com/biocore/qiime">GitHub</a>.</li>
</ul>

<hr />
<hr />
