<h4 id="assembling"><strong>1.</strong> Assembling</h4>

<p>At last!  All that filtering and diginorming is done, and we can get
down to the serious business of assembling.  Huzzah!</p>

<p>Let’s start with some collection of group files in <code>/mnt/assembly</code>.</p>

<h5 id="install-the-spades-assembler">Install the spades Assembler</h5>

<p>Let’s try assembling the sequences with the <a href="http://bioinf.spbau.ru/spades/">SPADes</a>.</p>

<p><code>
apt-get -y install cmake
</code></p>

<p>then:</p>

<p><code>
wget http://spades.bioinf.spbau.ru/release2.5.1/SPAdes-2.5.1.tar.gz
tar -xzf SPAdes-2.5.1.tar.gz
cd SPAdes-2.5.1
PREFIX=/usr/local ./spades_compile.sh
</code></p>

<h5 id="setting-up-the-spades-run">Setting Up The SPAdes run</h5>

<p>One of the reasons we are using SPAdes is because it runs an assembly on a set of k-mer values automatically.  It also has a simple structure.</p>

<p><code>
spades.py -s sequences.fastq -o assembly
</code></p>

<p>This will run for a while!  (We’ve already done this and will look at the assembly now).</p>

<h4 id="getting-stats-for-the-assemblies">Getting Stats For the Assemblies</h4>

<p>To get some basic stats for the assemblies, run:</p>

<p><code>
python /usr/local/share/khmer/sandbox/assemstats3.py 1000 assembly/scaffolds.fasta
</code></p>

<p>This will yield something like:</p>

<p><code>
N       sum     max     filename
38      671957  83467   dn.21/contigs.fa
32      668918  83568   dn.23/contigs.fa
35      668509  83401   dn.25/contigs.fa
31      671843  83817   dn.27/contigs.fa
32      669104  83721   dn.29/contigs.fa
32      672735  84066   dn.31/contigs.fa
32      673102  83774   dn.33/contigs.fa
31      674629  83912   dn.35/contigs.fa
31      677446  84200   dn.37/contigs.fa
33      681099  84554   dn.39/contigs.fa
35      685245  84852   dn.41/contigs.fa
40      686733  85276   dn.43/contigs.fa
41      649574  62719   nodn.21/contigs.fa
39      639388  62155   nodn.23/contigs.fa
49      646132  62145   nodn.25/contigs.fa
39      647100  83798   nodn.27/contigs.fa
38      650487  83750   nodn.29/contigs.fa
33      649863  83770   nodn.31/contigs.fa
31      636979  83822   nodn.33/contigs.fa
35      645536  83856   nodn.35/contigs.fa
36      647848  83800   nodn.37/contigs.fa
33      654660  83934   nodn.39/contigs.fa
36      645126  83897   nodn.41/contigs.fa
34      660289  83231   idba.dn.d/scaffold.fa
45      666147  41120   idba.nodn.d/scaffold.fa
</code></p>

<h4 id="annotating-your-metagenome-with-prokka"><strong>2.</strong> Annotating your metagenome with Prokka</h4>

<h5 id="installing-prokka">Installing Prokka</h5>

<p>I like the <a href="http://www.vicbioinformatics.com/software.prokka.shtml">Prokka Annotation Software Program</a> to annotate your assembled metagenome.</p>

<p>We have to download and install a lot of stuff, though – estimated ~15-20 minutes.</p>

<p>First, we need to install BioPerl and NCBI BLAST+; because we’re using an Amazon EC2 ubuntu machine, we’ll use the Debian Linux package installer <code>apt-get</code>.</p>

<p><code>
apt-get update
apt-get -y install bioperl ncbi-blast+
</code></p>

<p>Now you want to download and unpack Prokka:</p>

<p><code>
cd /mnt
curl -O http://www.vicbioinformatics.com/prokka-1.7.tar.gz
tar xzf prokka-1.7.tar.gz 
curl -O http://www.vicbioinformatics.com/prokka-1.7.2
cp prokka-1.7.2 prokka-1.7/bin/prokka
</code></p>

<p>Prokka depends on a lot of other software, too; so we’ll need to install those tools:</p>

<p>Install <a href="http://hmmer.janelia.org">HMMER</a>:</p>

<p><code>
cd /mnt
curl -O ftp://selab.janelia.org/pub/software/hmmer3/3.1b1/hmmer-3.1b1.tar.gz
tar xzf hmmer-3.1b1.tar.gz 
cd hmmer-3.1b1/
./configure --prefix=/usr &amp;&amp; make &amp;&amp; make install
</code></p>

<p>Install <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC373265">Aragorn</a>:</p>

<p><code>
cd /mnt
curl -O http://mbio-serv2.mbioekol.lu.se/ARAGORN/Downloads/aragorn1.2.36.tgz
tar -xvzf aragorn1.2.36.tgz
cd aragorn1.2.36/
gcc -O3 -ffast-math -finline-functions -o aragorn aragorn1.2.36.c
cp aragorn /usr/local/bin
</code></p>

<p>Install <a href="http://prodigal.ornl.gov">Prodigal</a>:</p>

<p><code>
cd /mnt
curl -O http://prodigal.googlecode.com/files/prodigal.v2_60.tar.gz
tar xzf prodigal.v2_60.tar.gz 
cd prodigal.v2_60/
make
cp prodigal /usr/local/bin
</code></p>

<p>Install <a href="http://www.ncbi.nlm.nih.gov/genbank/tbl2asn2">tbl2asn</a>:</p>

<p><code>
cd /mnt
curl -O ftp://ftp.ncbi.nih.gov/toolbox/ncbi_tools/converters/by_program/tbl2asn/linux64.tbl2asn.gz
gunzip linux64.tbl2asn.gz 
mv linux64.tbl2asn tbl2asn
chmod +x tbl2asn
cp tbl2asn /usr/local/bin
</code></p>

<p>Install <a href="&lt;http://www.biostars.org/p/63816/">GNU Parallel</a>:</p>

<p><code>
cd /mnt
curl -O http://ftp.gnu.org/gnu/parallel/parallel-20130822.tar.bz2
tar xjvf parallel-20130822.tar.bz2
cd parallel-20130822/
ls
./configure &amp;&amp; make &amp;&amp; make install
</code></p>

<p>Install <a href="http://infernal.janelia.org/">Infernal</a>:</p>

<p><code>
cd /mnt
curl -O http://selab.janelia.org/software/infernal/infernal-1.1rc4.tar.gz
tar xzf infernal-1.1rc4.tar.gz 
cd infernal-1.1rc4/
ls
./configure &amp;&amp; make &amp;&amp; make install
</code></p>

<h5 id="running-prokka-at-the-command-line">Running Prokka at the command line</h5>

<p>First we’ll use the khmer tool to remove all the sequences with ‘N’s in them (since prodigal fails if there are too many, and prokka uses prodigal):</p>

<p><code>
python /usr/local/share/khmer/sandbox/remove-N.py final-assembly.fa metagenome.fa
</code></p>

<p>Now, run Prokka::</p>

<p><code>
/mnt/prokka-1.7/bin/prokka metagenome.fa --outdir annotation --prefix assembly --metagenome
</code></p>

<p>There will be a bunch of files in the directory <code>annotation</code>.  Probably the most interesting is <code>assembly/assembly.faa</code>, which will contain a set of
annotated protein sequences derived from the metagenome.</p>

<p>Let’s look at some of the files now!</p>

<h4 id="assessing-assembly-quality-with-quast"><strong>3.</strong> Assessing Assembly Quality with QUAST</h4>

<p>We’re going to try to get an idea of what our assembly looks like by using the <a href="http://bioinf.spbau.ru/quast">QUAST (Quality Assessment Tool for Genome Assemblies)</a>.</p>

<p>First we need to have some tools installed:</p>

<p><code>
sudo apt-get install python-matplotlib
</code></p>

<p><code>
cd ~
mkdir PROGRAMS
cd PROGRAMS
wget https://downloads.sourceforge.net/project/quast/quast-2.3.tar.gz
tar xzvf quast-2.3.tar.gz
cd quast-2.3
</code></p>

<p>The program is installed here, but now we want to run a test to make sure:</p>

<p><code>
python metaquast.py --test
</code></p>

<p>A bunch of code will pass by, at the end you will see:</p>

<p>```
MetaQUAST finished.
Log saved to /home/ubuntu/PROGRAMS/quast-2.3/quast_test_output/metaquast.log</p>

<p>Finished: 2014-08-19 15:26:24
Elapsed time: 0:00:54.979271
Total NOTICEs: 12; WARNINGs: 0; non-fatal ERRORs: 0</p>

<p>Thank you for using QUAST!
```</p>

<p>…and it passed! Now we don’t have to do anything else to have a working QUAST.</p>

<p>We want to now do an assessment of our scaffolds:</p>

<p><code>
python metaquast.py -o folder ~/assembly/scaffolds.fasta
</code></p>

<p>Now we have a <code>folder</code> with our results (it’s pretty quick, which is a big plus after the rest of the tools we’ve used).</p>

<p>Let’s look at the files:</p>

<p><code>
cd folder/quast_output
ls -la
</code></p>

<p>Now you will see:</p>

<p><code>
total 96
drwxrwxr-x 4 ubuntu ubuntu  4096 Aug 19 15:37 .
drwxrwxr-x 4 ubuntu ubuntu  4096 Aug 19 15:37 ..
drwxrwxr-x 2 ubuntu ubuntu  4096 Aug 19 15:37 basic_stats
-rw-rw-r-- 1 ubuntu ubuntu  2105 Aug 19 15:37 quast.log
-rw-rw-r-- 1 ubuntu ubuntu 23161 Aug 19 15:37 report.html
drwxrwxr-x 6 ubuntu ubuntu  4096 Aug 19 15:37 report_html_aux
-rw-rw-r-- 1 ubuntu ubuntu 25569 Aug 19 15:37 report.pdf
-rw-rw-r-- 1 ubuntu ubuntu   799 Aug 19 15:37 report.tex
-rw-rw-r-- 1 ubuntu ubuntu   251 Aug 19 15:37 report.tsv
-rw-rw-r-- 1 ubuntu ubuntu   677 Aug 19 15:37 report.txt
-rw-rw-r-- 1 ubuntu ubuntu   704 Aug 19 15:37 transposed_report.tex
-rw-rw-r-- 1 ubuntu ubuntu   251 Aug 19 15:37 transposed_report.tsv
-rw-rw-r-- 1 ubuntu ubuntu   547 Aug 19 15:37 transposed_report.txt
</code></p>

<p>We can now look at the HTML report or the PDF report.  You can <code>scp</code> these files, or use a method to move files to your computer.</p>

<p>Let’s look at the assembly stats.  </p>

